db.go

```diff
@@ -0,0 +1,171 @@
+		altNamesRaw = []byte(altNames.Value)
+		return altNamesRaw, nil
+	}
+	altNamesRaw = []byte(altNames.Value)
+	if dbErr := d.Where(&DomainAltname{Domain: domain}).First(&altNames).Error; dbErr != nil {
+		return altNamesRaw, errors.Wrapf(dbErr, "Unable to get alternate domains for %s", domain)
+	}
+	altNamesRaw = []byte(altNames.Value)
+	return altNamesRaw, nil
+}
+
+/**
+Get User Info via email
+*/
+func getUserInfo(email string, d *gorm.DB) ([]byte, error) {
+	var userInfo UserInfo
+	var userInfoRaw []byte
+	if d.Where(&UserInfo{Email: email}).Find(&userInfo).RecordNotFound() {
+		userInfoRaw = []byte(userInfo.Value)
+		return userInfoRaw, nil
+	}
+	if dbErr := d.Where(&UserInfo{Email: email}).Find(&userInfo).Error; dbErr != nil {
+		return userInfoRaw, errors.Wrapf(dbErr, "Unable to get user info for %s", email)
+	}
+	userInfoRaw = []byte(userInfo.Value)
+	return userInfoRaw, nil
+}
+
+/**
+Get Certificate Details via Domain
+*/
+func getCertDetails(domain string, d *gorm.DB) ([]byte, error) {
+	var certDetails CertDetail
+	var certDetailsRaw []byte
+	if d.Where(&CertDetail{Domain: domain}).First(&certDetails).RecordNotFound() {
+		certDetailsRaw = []byte(certDetails.Value)
+		return certDetailsRaw, nil
+	}
+	if dbErr := d.Where(&CertDetail{Domain: domain}).First(&certDetails).Error; dbErr != nil {
+		return certDetailsRaw, errors.Wrapf(dbErr, "Unable to get certificate details for %s", domain)
+	}
+	certDetailsRaw = []byte(certDetails.Value)
+	return certDetailsRaw, nil
+}
+
+/**
+Save User Information key email, value userInfo
+*/
+func addUserInfo(email string, userInfo []byte, d *gorm.DB) error {
+	s := string(userInfo)
+	if dbErr := d.Create(&UserInfo{Email: email, Value: s}).Error; dbErr != nil {
+		return errors.Wrapf(dbErr, "Unable to add user info for %s", email)
+	}
+	return nil
+}
+
+/**
+Save Certificate Details key domain, value certDetails
+*/
+func addCertDetails(domain string, certDetails []byte, d *gorm.DB) error {
+	s := string(certDetails)
+	if dbErr := d.Create(&CertDetail{Domain: domain, Value: s}).Error; dbErr != nil {
+		return errors.Wrapf(dbErr, "Unable to add cert details for %s", domain)
+	}
+	return nil
+}
+
+/**
+Save Alt Names Details key domain, value altNames
+*/
+func addAltNames(domain string, altNames []byte, d *gorm.DB) error {
+	s := string(altNames)
+	if dbErr := d.Create(&DomainAltname{Domain: domain, Value: s}).Error; dbErr != nil {
+		return errors.Wrapf(dbErr, "Unable to add alt names for %s", domain)
+	}
+	return nil
+}
+
+/**
+Update Alt Names Details key domain, new values altNames
+*/
+func updateAltNames(domain string, altNamesRaw []byte, d *gorm.DB) error {
+	s := string(altNamesRaw)
+	var altNames DomainAltname
+	if dbErr := d.Where(&DomainAltname{Domain: domain}).First(&altNames).Error; dbErr != nil {
+		return errors.Wrapf(dbErr, "Unable to update alt names for %s", domain)
+	}
+	altNames.Value = s
+	d.Save(&altNames)
```

Comment:
You need to check the Save call for errors as well.

Welcome to go :trollface:.

Reply:
doh!